plumbing/transport/server/loader.go

```diff
@@ -0,0 +1,51 @@
+package server
+
+import (
+	"gopkg.in/src-d/go-git.v4/plumbing/storer"
+	"gopkg.in/src-d/go-git.v4/plumbing/transport"
+	"gopkg.in/src-d/go-git.v4/storage/filesystem"
+
+	"srcd.works/go-billy.v1"
+	"srcd.works/go-billy.v1/os"
+)
+
+// DefaultLoader is a filesystem loader ignoring host and resolving paths to /.
+var DefaultLoader = NewFilesystemLoader(os.New("/"))
+
+// Loader loads repository's storer.Storer based on an optional host and a path.
+type Loader interface {
+	// Load loads a storer.Storer given a transport.Endpoint.
+	// Returns transport.ErrRepositoryNotFound if the repository does not
+	// exist.
+	Load(ep transport.Endpoint) (storer.Storer, error)
+}
+
+type fsLoader struct {
+	base billy.Filesystem
+}
+
+// NewFilesystemLoader creates a Loader that ignores host and resolves paths
+// with a given base filesystem.
+func NewFilesystemLoader(base billy.Filesystem) Loader {
+	return &fsLoader{base}
+}
+
+func (l *fsLoader) Load(ep transport.Endpoint) (storer.Storer, error) {
+	fs := l.base.Dir(ep.Path)
+	if _, err := fs.Stat("config"); err != nil {
+		return nil, transport.ErrRepositoryNotFound
+	}
+
+	return filesystem.NewStorage(fs)
+}
+
+type MapLoader map[transport.Endpoint]storer.Storer
```

Comment:
I don't think I can provide a better name for `Load` right now.

I don't know what problems are you trying to solve with this `loader` thing. The description of the PR is not enough to understand why a `Loader` is needed and how we were doing that before having a `loader`. I think I am too disconnected from the project to give my opinion.

From my point of view: I don't understand some of the names here, they are too generic:

- 'storer.Storer': of what?, you say it is actually a repository in the previous answer but the name does not give any hints about it. By its definition it is anything capable of holding encoded objects and references (which looks as a classical git repository on disk to me, this is a packfile + references).

- 'server.Loader': of what?, it does not actually loads a server as its name implies, but apparently it is a database capable of returning the encoded objects and references of a repository from its endpoint. There is also a comment still says it looks at an (optional) host and path, that may be outdated. It looks as a resolver to me: given the name of a resource, returns the resource.

I don't think any of these is wrong, it is just that I am very lost and I have not found a description of the problem and how we have decided to solve it, so it is very difficult to understand your solution and think about it.